### ISSUE TITLE
Comment in the beginning of generated files

### ISSUE THREAD

**Issue by emilwihlander** [dotansimha/graphql-code-generator#5063]
**Is your feature request related to a problem? Please describe.**
No indication is given that a file is generated inside the file itself with default settings. I think that this should be the standard to clearly indicate to all developers that a file is generated.

**Describe the solution you'd like**
Always prepend generated files with a block comment describing that the file has been generated by graphql-code-generator, maybe even which plugins and what config that were used (This might be a security issue, since it might expose version/plugin-related vulnerabilities?).

Something like:
```typescript
/*
Generated by @graphql-codegen/cli@1.19.1
Plugins:
    @graphql-codegen/typescript@1.17.11
    @graphql-codegen/typescript-resolvers@1.17.10
Config:
    federation: true
    useIndexSignature: true
*/
```

**Describe alternatives you've considered**
At the moment it is possible, with the help of the add-plugin, to prepend comments describing that the file is generated. This isn't optimal though.

**Additional context**

**Comment by maapteh** [comment]
Why is it not optimal. Our team for example start the filename with generated, and we need the eslint disabling in the top and also add the info about autogenerating for the users we call developers with just this add block :)
Personally i think add plugin gives full flexibility.

**Comment by dotansimha** [comment]
I agree with @maapteh on that. Maybe we can offer a "built-in" plugin called `debug` or something similar, to print those (you can also create it, should be pretty simple).

**Comment by grelas** [comment]
@emilwihlander fyi I was able to use the `@graphql-codegen/add` plugin to prepend a comment at the top of the generated file.

**Comment by eddeee888** [comment]
Our recommendation is using `@graphql-codegen/add` to prepend this on top of files.
If you need explicit version, you could parse `package.json` in [codegen.ts](https://the-guild.dev/graphql/codegen/docs/config-reference/codegen-config) and use it with the `add` plugin.

Otherwise, a local/customised plugin can work as well 🙂
